### PR TITLE
Pickling Weights

### PR DESCRIPTION
**Pull request by bachlava**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -16,22 +17,29 @@ def plot_history(history):
...
-    history = model.fit(
-        train_data,
-        validation_data=test_data,
-        epochs=config.MODEL_NUM_EPOCHS,
-        batch_size=config.MODEL_BATCH_SIZE,
-        workers=config.MODEL_WORKERS,
-    )
-    plot_history(history)
+    if has_saved_weights():
+        load_status = model.load_weights(f'./{config.PICKLE_PATH}/ckpt')
```
Comment (semibran) [inline main.py]: EXTREME nit but can we remove the relative path and use a join here to be consistent. can make a `load_weights(model, checkpoint_name)` function to cut down on the imports in main

File: main.py [hunk 2]
```diff
@@ -16,22 +17,29 @@ def plot_history(history):
...
+    else:
+        history = model.fit(
+            train_data,
+            validation_data=test_data,
+            epochs=config.MODEL_NUM_EPOCHS,
+            batch_size=config.MODEL_BATCH_SIZE,
+            workers=config.MODEL_WORKERS,
+            use_multiprocessing=True
+        )
+        serialize_weights(model, 'ckpt')
```
Comment (semibran) [inline main.py]: `'ckpt'` can be a constant here and/or a default arg for `checkpoint_name`

File: configs/model.py [hunk 3]
```diff
@@ -1,12 +1,13 @@
 DATASET_PATH = 'dataset'
 OUTPUT_PATH = 'dist'
+PICKLE_PATH = 'pickle'
```
Comment (semibran) [inline configs/model.py]: what artifacts does pickling make? could just write the weights to `dist/`?

**Review by semibran — COMMENTED** [review]
Look at my code reviewers dawg this project is never gonna get done

File: configs/model.py (hunk 3 above)
Comment (bachlava) [reply configs/model.py]: these bad boys, so basically preference on where to store them

image

**Review by semibran — APPROVED** [review]

**Pull request merged**